docs/specifications/tests/Zone-TP/zone01.md

```diff
@@ -3,43 +3,262 @@
+   1. If the SOA MNAME name server name is not part of the
+      *Child Nameservers* set, then output *[Z01_MNAME_NOT_IN_NS_LIST]* with
+      SOA MNAME name server name.
+   2. Do a [DNS Lookup] of SOA MNAME name server name (A and AAAA) and add the
+      SOA MNAME name server IP address(es) found to the *MNAME Nameservers* set
+      for the same entry as the SOA MNAME name server name.
+   3. If at least one IP address from the previous step was found, then:
+      1. For each SOA MNAME name server IP for the SOA MNAME name server name do: 
+         1. If the IP address is a localhost address (127.0.0.1 or ::1), then output
+            *[Z01_MNAME_HAS_LOCALHOST_ADDR]* with SOA MNAME name server name and IP.
+         2. Else, send *SOA Query* to the name server and collect the response.
+            1. If there is a DNS response, with [RCODE Name] "NoError" and
+               an SOA record in the answer section, then:
+               1. If the AA flag is not set, then output *[Z01_MNAME_NOT_AUTHORITATIVE]* 
+                  with SOA MNAME name server name and IP address, and name server(s) IP
+                  address(es) from the *SERIAL Nameservers* set.
+               2. Else, add the SOA SERIAL field to the *MNAME Nameservers* set
+                  for the SOA MNAME name server name and IP pair.
+            2. Else if [RCODE Name] is not "NoError", then output
+               *[Z01_MNAME_UNEXPECTED_RCODE]* with [RCODE Name], SOA MNAME name server name and
+               IP address, and name server(s) IP address(es) from the *SERIAL Nameservers* set.
+            3. Else if there is no SOA record in the answer section, then output
+               *[Z01_MNAME_MISSING_SOA_RECORD]* with SOA MNAME name server name and IP
+               address, and name server(s) IP address(es) from the *SERIAL Nameservers* set.
+            4. Else, output *[Z01_MNAME_NO_RESPONSE]* with SOA MNAME name server name and IP
+               address, and name server(s) IP address(es) from the *SERIAL Nameservers* set.
+         3. Go to next SOA MNAME name server IP.
+   4. Else, output *[Z01_MNAME_NOT_RESOLVE]* with SOA MNAME name server name,
+      and name server(s) IP address(es) from the *SERIAL Nameservers* set.
+   5. Go to next SOA MNAME name server name.
+
+10. If there is no SOA SERIAL in the *MNAME Nameservers* set, then
+    terminate these procedures.
+
+11. For each SOA SERIAL in *MNAME Nameservers* do:
+    1. For each SOA SERIAL in *SERIAL Nameservers* do:
+       1. Compare both SOA SERIAL values (using the arithmetic in [RFC1982]).
+          1. If the one from *SERIAL Nameservers* is higher, then add SOA MNAME name
+             server name and IP address to the *MNAME Not Master* set, and go to next
+             SOA SERIAL (from the *MNAME Nameservers* set).
+          2. Go to next SOA SERIAL.
+    2. Add SOA MNAME name server name and IP address to the *MNAME Is Master* set.
+    3. Go to next SOA SERIAL.
```

Comment:
In 11.1.1 the two SOA SERIALs are compared. If the one from *SERIAL Nameservers* is higher then the loop goes to the next SOA SERIAL from *MNAME Nameservers* (11.1.1.1), else the loop goes to the next SOA SERIAL from *SERIAL Nameservers* (11.1.1.2). I do not see how you come to 11.1.2.

I suggest that 11.1.1.2 is changed from "Go to next SOA SERIAL" to "Else, go to next SOA SERIAL (from the *SERIAL Nameservers* set)".

I think 11.1.3 could be removed.

Reply:
There is no 11.1.2. I assume you talk about 11.2. You go to 11.2 when you keep going to 11.1.1.2, during the inner loop in 11.1. You restart from 11.1 when you reach 11.1.1.1 (I think the confusion comes from that, in my mind this is the logic I use), with the next *MNAME Namservers* SERIAL value. Thus 11.1.3 is also necessary to loop through all values in the *MNAME Nameservers* set (step 11).

Is it unclear this way? Should I try and reword it all?